tests/csapi/room_versions_test.go

```diff
@@ -0,0 +1,193 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestRoomVersions(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintFederationTwoLocalOneRemote)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs1", "@bob:hs1")
+	charlie := deployment.Client(t, "hs2", "@charlie:hs2")
+
+	roomVersions := gomatrixserverlib.RoomVersions()
+
+	t.Run("Parallel", func(t *testing.T) {
+		// iterate over all room versions
+		for v := range roomVersions {
+			roomVersion := v
+			// sytest: User can create and send/receive messages in a room with version $version
+			t.Run(fmt.Sprintf("User can create and send/receive messages in a room with version %s", roomVersion), func(t *testing.T) {
+				t.Parallel()
+				roomID := createRoomSynced(t, alice, map[string]interface{}{
+					"room_version": roomVersion,
+				})
+				alice.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(alice.UserID, roomID))
```

Comment:
Isn't the whole point of the `createRoomSynced` function to literally do this? Why do it again?